### ISSUE TITLE
Google Voice bypasses all VPN apps, including AdGuard VPN

### ISSUE THREAD

**Issue by EntropySmoke** [AdguardTeam/AdGuardVPNForWindows#43]
### Prerequisites
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details
- **AdGuard VPN version:**
0.7.1
- **VPN mode:**
AdGuard VPN in VPN Mode, AdGuard AdBlocker in Local Proxy Auto-Mode
- **Operating system and version:**
Android 10 (rooted Lineage OS 17.1), no carrier/SIM card/data, WiFi/Bluetooth/USB-only connections
- **Other network-level software:**
AdGuard Home DNS server, AFWall+, custom IPTables scripts

### Expected Behavior
The expectation is that all apps would be tunneled through AdGuard VPN tunnel.

### Actual Behavior
TCP dumps show that Google Voice bypasses all VPN apps, including AdGuard VPN and OpenVPN Connect. Google Voice routes itself through Tun0 VPN adapter 99% of the time, but **per each received or made call** Google Voice always creates at least one direct connection over STUN UDP ports 19302-19305, which exposes both public and local IP. I can submit my TCPDump log, but privately. 

AdGuard AdBlocker in Local Proxy Auto-Mode log doesn't register that connection because Local Auto-Proxy Mode does not filter UDP packets. AdGuard AdBlocker's "Disable WebRTC" is enabled and works in almost all apps, but doesn't work for Google Voice...

AdGuard AdBlocker Filter Log detects Google Voice as "root" (instead of "Google Voice") when Google Voice makes a TCP connection to 216.239.36.145 over port 443 right before making and/or receiving a call. There are no other apps that register themselves as "root" in AdGuard Filter Log, but I don't know if this is important.

This behavior happens regardless of whether I use default IPTable rules/chain rules and/or AdGuard AdBlocker. 

### Additional Information
This affects all Android VPN apps, regardless of VPN protocol and regardless of DNS resolution settings, but AdGuard VPN could be the first one to make Google Voice calls be fully tunneled via VPN.

**Comment by ameshkov** [comment]
Issue moved to [AdguardTeam/AdGuardVPNForAndroid #64](https://github.com/AdguardTeam/AdGuardVPNForAndroid/issues/64) via [**ZenHub**](https://www.zenhub.com/)

**Comment by EntropySmoke** [comment]
Using netstat command during a Google Voice call shows that domain resolution for 216.239.36.145 (Google Voice) also bypasses VPN Tun0 adapter and connects directly with my public IP. Aside from Google Voice, the one and only time netstat shows the same result is when my devices are trying to establish an initial connection to VPN services. That is normal, but Google Voice bypassing VPN connection like that is strange... 

If Google Voice is not installed, AdGuard AdBlocker can block 216.239.36.145 via DNS or User rules, but if you install Google Voice, the same rule stops working and Google Voice registers itself as "root" in AdGuard AdBlocker.

**Comment by ameshkov** [comment]
Well, actually this is rather strange. It's okay for an ISP to bypass when they're using Wi-Fi calling, but Google Voice is supposed to be just an app.

Check `pref.excluded.packages` in low-level settings, maybe Google Voice uses one of their uids?

**Comment by EntropySmoke** [comment]
Google Voice definitely doesn't recognize AdGuard's HTTPS certificates, even on rooted phones with a Magisk module that moves user certificates to system directory. 

What is also strange is that the Google Voice STUN connection that leaks local LAN/WiFi IP shows up when running "tcpdump -i tun0" , but not "tcpdump -i wlan0" command.  

Android Logcat shows a JavaScript error every time I make and/or receive a Google Voice call.

**Comment by ameshkov** [comment]
> What is also strange is that the Google Voice STUN connection that leaks local LAN/WiFi IP shows up when running "tcpdump -i tun0" , but not "tcpdump -i wlan0" command.

Hm, wait, so do you use AdGuard + AdGuard VPN, right?

This may be the reason. I might be wrong, but maybe ports that aren't included in `pref.filtered.ports` aren't routed via AdGuard VPN in this case.

@artemiv4nov could you confirm this?

**Comment by EntropySmoke** [comment]
There was the same result for all tests I conducted:
1. AdGuard AdBlocker Local VPN (by itself)
2. AdGuard AdBlocker Local VPN + AdGuardVPN-Proxy/NordVPN-Proxy
3. AdGuard AdBlocker (HTTP Local Proxy Auto-Mode) + AdGuardVPN/NordVPN/OpenVPN-Connect
4. AdGuardVPN/NordVPN/OpenVPN-Connect by itself

I tried using 1:65535 port range in AdGuard AdBlocker, but the STUN UDP Port 19302-19305 connections didn't show up in any logs, except for "LOGCAT" and "TCPDump -i tun0", but with local wlan0 IP instead of tun0 IP.

TCP connections to 216.239.36.145, voice.google.com, google.com, stun.l.google.com, and lens.l.google.com showed up in AdGuard AdBlocker log, but those connections never leaked local IP in neither tun0 nor wlan0 (or any other) interfaces.
